version: '3.3'

volumes:
  postgres_data:
  django_static:

services:
  postgres:
    image: postgres:15.6
    ports:
      - "5432:5432"
    env_file: .env
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready" ]
      interval: 5s
      timeout: 5s
      retries: 10
    depends_on:
      - redis
  redis:
    image: redis:7.2.4
  django:
    build:
      context: .
      dockerfile: ./web/Dockerfile
    env_file: .env
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - django_static:/django_static/
  celery:
    build:
      context: .
      dockerfile: ./celery/Dockerfile
    env_file: .env
    depends_on:
      - postgres
      - redis
      - django
    volumes:
      - /:/outer/
  celery-beat:
    build:
      context: .
      dockerfile: ./celery-beat/Dockerfile
    env_file: .env
    depends_on:
      - postgres
      - redis
      - django
      - celery
  nginx:
    build: ./nginx/.
    ports:
      - "80:80"
    volumes:
      - django_static:/usr/share/nginx/html/
    depends_on:
      - django
